internals/daemon/daemon_test.go

```diff
@@ -67,9 +68,15 @@ func (s *daemonSuite) SetUpTest(c *C) {
 		s.notified = append(s.notified, notif)
 		return nil
 	}
+	if s.savedAPI == nil {
+		s.savedAPI = make([]*Command, len(API))
+		copy(s.savedAPI, API)
+	}
 }
 
 func (s *daemonSuite) TearDownTest(c *C) {
+	copy(API, s.savedAPI)
+	API = API[:len(s.savedAPI)]
```

Comment:
This logic is incorrect if len(API) < len(s.savedAPI). Also, it seems a bit wasteful to be doing this when this is only touched by one trivial test.  Can we please go back to the prior approach, and simply drop the one offending entry that the test is adding?  We don't even need to iterate... we can just panic if that last element isn't the expected one, and then drop the last element.

Reply:
Should be ok now.